src/time-wait-sync/time-wait-sync.c

```diff
@@ -33,18 +34,28 @@
 
 #include "fd-util.h"
 #include "missing.h"
+#include "mkdir.h"
 #include "signal-util.h"
 #include "time-util.h"
 
 typedef struct ClockState {
-        int fd;                        /* non-negative is descriptor from timerfd_create */
-        int adjtime_state;             /* return value from last adjtimex(2) call */
-        sd_event_source *event_source; /* non-null is the active io event source */
+        int timerfd_fd;                  /* non-negative is descriptor from timerfd_create */
+        int adjtime_state;               /* return value from last adjtimex(2) call */
+        sd_event_source *timerfd_event_source; /* non-null is the active io event source */
+        int notify_fd;
+        sd_event_source *notify_event_source;
```

Comment:
notify_fd sounds overly generic. I think inotify_fd would be a lot more explicit and only costs a single character more!